lesson01/assignment/scripts.js

```diff
@@ -2,28 +2,115 @@
 document. addEventListener("DOMContentLoaded", function(){
 
   // Find the example form, and handle its submit event
-  const exampleForm = document.querySelector('#example-form')
+  const form1 = document.querySelector('#form1')
 
-  exampleForm.addEventListener('submit', function(event){
+  form1.addEventListener('submit', function(event){
     // Prevent the form from trying to submit to a server
     event.preventDefault()
 
     // An array that we can transform based on user input
     let data = [0, 1, 2, 3, 5, 8]
 
-    const inputField = document.querySelector('#example-form .input')
+    const inputField = document.querySelector('#form1 .input')
     const userInput = inputField.value
     const userNumber = parseInt(userInput)
 
-    // If the user's input was a number, push it into the array and continue
     if(!isNaN(userNumber)){
-      data.push(userNumber)
+      
+      //1: Map Function we are going to add the favorite number by the user to each element in the array:
+      let mapArray = data.map(function(value) {
+        return value + userNumber;
+      });
 
-      // Use reduce to sum all the numbers in the data array
-      const sum = data.reduce((accumulator, currentElement) => accumulator + currentElement);
+      document.querySelector('#form1 .output').innerHTML = mapArray.toString();
+    }
+
+    // Clear the form field so the user can try again
+    inputField.value = ''
+  })
+
+  /****************************************************************************************************************************************/
+
+  const form2 = document.querySelector('#form2')
+
+  form2.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault()
+
+    // An array that we can transform based on user input
+    let data = [0, 1, 2, 3, 5, 8]
+
+    const inputField = document.querySelector('#form2 .input')
+    const userInput = inputField.value
+    const userNumber = parseInt(userInput)
+
+    if(!isNaN(userNumber)){
+      
+
+      //2: Filter Function, we are going get elements that are greater than the favorite number
+      let filteredArray = data.filter(function(value){
+        return value > userNumber;
+      });
+
+      document.querySelector('#form2 .output').innerHTML = filteredArray.toString();
+    }
+
+    // Clear the form field so the user can try again
+    inputField.value = ''
+  })
```

Comment:
Since these arrays are defined in the javascript, you might consider inserting them into the HTML dynamically, rather than hardcoding them into the comment, so they always stay in sync. 
For example, in HTML:
`<p>Filter Function, we are going to get elements that are greater than the favorite number in the <span class="array"></span> array <span class="output"></span></p>`

In Javascript, on line 35, so it's part of page ready event listener, not form submit event listener:
```
const form2Data = [0, 1, 2, 3, 5, 8]
form2.array.innerHTML = form2Data
```
Then on line 51, where the data array is used, refer to the form2Data variable rather than defining the array again.